iib/web/api_v1.py

```diff
@@ -481,10 +485,43 @@ def _get_user_queue(serial: Optional[bool] = False) -> Optional[str]:
     else:
         labeled_username = f'PARALLEL:{username}'
 
-    queue = flask.current_app.config['IIB_USER_TO_QUEUE'].get(labeled_username)
-    if not queue:
-        queue = flask.current_app.config['IIB_USER_TO_QUEUE'].get(username)
-    return queue
+    index_queue = flask.current_app.config['IIB_USER_TO_QUEUE'].get(labeled_username)
+    if not index_queue:
+        index_queue = flask.current_app.config['IIB_USER_TO_QUEUE'].get(username)
+
+    if index_queue is None:
+        return index_queue
+    elif all(isinstance(value, str) for value in index_queue.values()):
+        # keep original behavior for IIB_USER_TO_QUEUE of type dict[str,str])
+        return index_queue
+    elif all(isinstance(value, dict) for value in index_queue.values()):
+        queue_default = index_queue.get('all')
+        if not queue_default:
+            if not from_index_pull_spec:
+                # we do not have defined default queue and there is no pull spect
+                # for which we should obtain queue. Returning None as celery default
+                return None
+
+            error_msg = f"default queue (all) is not defined for user {username}"
+            logging.error(error_msg)
+            raise IIBError(error_msg)
```

Comment:
Should we fail or would it be better to return `None` as celery default?

Reply:
failing sounds okay to me

Reply:
Can I suggest not checking the structure of the dictionary here. At this point, we should assume the config is set correctly. We should probably add validation here: https://github.com/release-engineering/iib/blob/master/iib/web/app.py#L45 

There we can check if the structure is correct and if the default `all` is present.

Reply:
I will add the validation, that is good point. However I need to know how to load the config. So if we are using one variable supporting two formats, then I need to do this check on this place. (or call a function which determines the structure)

Reply:
We can add validation for both formats in app.py, no? Or we simply make this a breaking change?

Reply:
Yes - I have changes made but forget to push it. I will have it ready for review tomorrow.